prov/hook/hook_hmem/src/hook_hmem.c

```diff
@@ -1663,6 +1663,12 @@ static int hook_hmem_mr_regv(struct fid *fid, const struct iovec *iov,
 	attr.auth_key_size = 0;
 	attr.auth_key = NULL;
 
+	/* hook_hmem_mr_regattr will determine the correct iface,
+	 * this silences warnings about uninitialized fields
+	 */
+	attr.iface = FI_HMEM_SYSTEM;
+	attr.device.reserved = 0;
```

Comment:
We do require C99, so can initialize it up in L1655 as `struct fi_mr_attr attr = {0};` too.